frontend/src/metabase-types/api/query.ts

```diff
@@ -355,7 +355,8 @@ export type Expression =
       ExpressionOperand,
       ExpressionOperand,
       ExpressionOperand,
-    ];
+    ]
+  | ConcreteFieldReference;
```

Comment:
I think we're missing this type. As the screenshot below shows, an expression could be a field reference or a reference to another expression.

![Screenshot 2023-09-07 at 11 17 25 PM](https://github.com/metabase/metabase/assets/1937582/d2de32a3-e222-479d-9090-1758d3d6330e)

https://github.com/metabase/metabase/blob/a90511e342ea8b90455b7ead2ba3967780ef6f19/frontend/src/metabase-types/api/query.ts#L290-L297

I use it [in the test](https://github.com/metabase/metabase/pull/33810/files#diff-dbaaca4e4bf6f621873aa7622e48a9d5a7b1f7df0272354b12d7c2f406670a41R78-R80), otherwise, TypeScript would complain the type is incorrect.